### DISCUSSION TITLE
Plan more forward

### DISCUSSION THREAD

**Discussion by zipus** [frePPLe/frepple#633] [Q&A]
I'm using plan.administrativeLeadtime=999 to plan forward. But I still have some issues with the result.

I created a simple calendar to illustrate better my problem.
![imagen](https://github.com/frePPLe/frepple/assets/1032986/33b9a29a-7c7e-4301-b762-75aacaf36454)

The thing is that Resources 2, and 3 are free to use from day 5. But frepple waits those operations until just starting the last operation. In our use case we can start on day 5 using resource 2, so if new order comes in on day 7 and needs the same resource maybe I can allocate on the 8th day instead of 9th.

Is this supported in Frepple? Any parameter I can adjust?

**Comment by jdetaeye** [comment]
The parameter plan.administrativeLeadtime works on the demand level.  We plan for an ASAP delivery, ahead of the due date. 
It's possible that the plan doesn't plan all operations ASAP.

For instance, the final assembly operation can only be scheduled on day 20 because material or capacity are completely missing before that time.  The production of the subassemblies will be scheduled to finish the subassemblies just before day 20, even if all material and capacity is available to start today.

You can look into  the field "operation.posttime" https://frepple.com/docs/current/model-reference/operations.php to build in extra buffer time between operations.  The buffer time is a soft constraint - ie the algorithm tries to respect it, but a plan with a smaller time between the operations is ok.

In the example above setting the posttime to 999 days for the subassemblies should plan these earlier. 
Rather than setting it to 999 days, I'd recommend to set it to eg 7 days or so.   This will pull the operations a bit early, but avoid producing excessively up excessive stocks in the factory.
